sionna/channel/optical/fiber.py

```diff
@@ -0,0 +1,309 @@
+#
+# SPDX-FileCopyrightText: Copyright (c) 2021-2022 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Author: Tim Alexander Uhlemann <uhlemann@ieee.org>
+
+"""
+This module defines the split-step Fourier method to approximate the solution of
+the nonlinear Schroedinger equation.
+"""
+
+import tensorflow as tf
+from tensorflow.keras.layers import Layer
+import sionna
+from sionna.channel import utils
+
+
+class SSFM(Layer):
+    # pylint: disable=line-too-long
+    r"""SSFM(alpha, beta_2, f_c, gamma, half_window_length, ell, n_ssfm, n_sp, dt, t_norm, with_amplification, with_attenuation, with_dispersion, with_nonlinearity, swap_memory=True, dtype=tf.complex64, **kwargs)
+
+    Layer implementing the split-step Fourier method (SSFM).
+
+    The SSFM (first mentioned in [SSFM]_) is implemented as
+    symmetrized SSFM according to Eq. (7) of [SymSSFM]_ and can be written as
+
+    .. math::
+
+        E(z+\Delta_z,t) \approx \exp\left(\frac{\Delta_z}{2}\hat{D}\right)\exp\left(\int^{z+\Delta_z}_z \hat{N}(z')dz'\right)\exp\left(\frac{\Delta_z}{2}\hat{D}\right)E(z,\,t)
+
+    The integral is approximated by :math:`\Delta_z\hat{N}`.
+
+    Additionally, ideally distributed Raman amplification may be applied, which
+    is implemented as in [RamanASE]_.
+
+    The SSFM operates on normalized time :math:`T_\text{norm}` (e.g., 1 ps) and
+    distance units :math:`L_\text{norm}` (e.g., 1 km). Hence, parameters as well
+    as the signal itself have to be given in the same normalized units. Note
+    that, despite the normalization, the SSFM is implemented with physical units
```

Comment:
This explanation about units is unclear and needs to be improved.

Reply:
Tried to reformulate it - still in progress.